modules/wfo.js

```diff
@@ -0,0 +1,171 @@
+// rss and rss2 don't show the author, thus using type=atom.
+const URI = 'https://wetfishonline.com/forum/index.php?action=.xml;sa=recent;limit=10;type=atom';
+
+const util = require('util');
+var FeedParser = require('feedparser');
+var request = require('request');
+
+////////////////////////////////////////////////////////////////////////////////
+function State() {
+    this.STATE_NOTREADY = 0; // Before we have seen_guids, so we don't know what msgs are new.
+    this.STATE_NORMAL = 1;   // We're active.
+
+    this.TIMEOUT = 5 * (60 * 1000);
+    this.LIMIT = 2; // Max num items we would write about each time?
+
+    this.seen_guids = []; // TODO: Clean up old items.
+    this.current_state = this.STATE_NOTREADY;
+}
+State.prototype.getCurrentTimeout = function() {
+    if (this.current_state == this.STATE_NOTREADY) {
+        return 10000; // This is used when there's an error before we're ready.
+    } else if (this.current_state == this.STATE_NORMAL) {
+        return this.TIMEOUT;
+    }
+    console.log('getCurrentTimeout error');
+}
+State.prototype.newItems = function(items, announce_fn) {
+
+    var need_announce = [];
+
+    // my nodejs isn't ES6 yet so i can't use the fat arrow function
+    // defn.
+    var thisstate = this;
+
+    items.forEach(function (item) {
+        if (thisstate.seen_guids.indexOf(item.guid) == -1) {
+            need_announce.push(item);
+            thisstate.seen_guids.push(item.guid);
+        }
+    });
+
+    if (this.current_state == this.STATE_NORMAL) {
+        // feed gives newest entries first, but we should
+        // announce oldest first.
+        need_announce.reverse();
+
+        if (need_announce.length >= this.LIMIT) {
+
+            announce_fn(util.format("9[WFO] At least9 %d posts were made in the last %d minutes! Join the action at 9https://wetfishonline.com/",
```

Comment:
I think it should only say "at least" if there were more changes than can be seen in a single page of RSS results. By default it should say specifically how many posts there are.